src/domain/synthetics/tokens/use24Volumes.ts

```diff
@@ -0,0 +1,69 @@
+import { gql } from "@apollo/client";
+import useSWR from "swr";
+import type { Address } from "viem";
+
+import { selectChainId } from "context/SyntheticsStateContext/selectors/globalSelectors";
+import { useSelector } from "context/SyntheticsStateContext/utils";
+
+import { NATIVE_TOKEN_ADDRESS, convertTokenAddress } from "config/tokens";
+import { TIMEZONE_OFFSET_SEC } from "domain/prices/constants";
+import { getSubsquidGraphClient } from "lib/subgraph/clients";
+
+type PositionVolumeInfosResponse = Record<Address, bigint>;
+
+const MARKET_VOLUMES_QUERY = gql`
+  query MarketVolumesInfoResolver($indexTokenAddresses: [String!]!, $timestamp: Float!) {
+    positionsVolume24hByIndexTokens(where: { timestamp: $timestamp, indexTokenAddresses: $indexTokenAddresses }) {
+      volume
+      indexTokenAddress
+    }
+  }
+`;
+
+export function use24hVolumes(indexTokenAddresses: (Address | undefined)[]) {
```

Comment:
I see you used new resolver in squid which can return map of markets by index token. There's already `positionsVolume` resolver which return a list of volumes for every market. Also we fetch volume for single market in `use24hVolume`, i'd like to decrease an amount of different approaches here.